

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>OpenSMOG &mdash; OpenSMOG 0.0.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/jupyter-sphinx.css" type="text/css" />
  <link rel="stylesheet" href="_static/thebelab.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/thebelab-helper.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="https://unpkg.com/@jupyter-widgets/html-manager@^0.20.0/dist/embed-amd.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "document", "processHtmlClass": "math|output_area"}}</script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Using SMOG2 - C-alpha and All-Atoms" href="Tutorials/SMOG2_usage.html" />
    <link rel="prev" title="Introduction" href="GettingStarted/intro.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> OpenSMOG
          

          
          </a>

          
            
            
              <div class="version">
                0.0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="GettingStarted/install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="GettingStarted/intro.html">Introduction</a></li>
</ul>
<p class="caption"><span class="caption-text">OpenSMOG Modules</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">OpenSMOG</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Tutorials/SMOG2_usage.html">Using SMOG2 - C-alpha and All-Atoms</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials/SBM_CA.html">C-alpha Structure-based Model simulation using OpenSMOG</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials/SBM_AA.html">All-Atoms Structure-based Model simulation using OpenSMOG</a></li>
</ul>
<p class="caption"><span class="caption-text">Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Reference/citing.html">How to cite <strong>OpenSMOG</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="Reference/references.html">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="Reference/license.html">License</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">OpenSMOG</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>OpenSMOG</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/OpenSMOG.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="module-OpenSMOG">
<span id="opensmog"></span><h1>OpenSMOG<a class="headerlink" href="#module-OpenSMOG" title="Permalink to this headline">¶</a></h1>
<p>The <code class="xref py py-class docutils literal notranslate"><span class="pre">OpenSMOG</span></code> classes perform molecular dynamics using Structure-Based Models (SBM) for biomolecular simulations.
<code class="xref py py-class docutils literal notranslate"><span class="pre">OpenSMOG</span></code> uses force fields generated by SMOG 2, and it allows the simulations of a wide variety of potential forms, including commonly employed C-alpha and all-atom variants.
Details about the default models in SMOG 2 can be found in:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>SMOG server</strong>: <a class="reference external" href="https://smog-server.org/">https://smog-server.org/</a></p></li>
<li><p><strong>C-alpha</strong>: Clementi, C., Nymeyer, H. and Onuchic, J.N., 2000. Topological and energetic factors: what determines the structural details of the transition state ensemble and “en-route” intermediates for protein folding? An investigation for small globular proteins. Journal of molecular biology, 298(5), pp.937-953.</p></li>
<li><p><strong>All-Atom</strong>: Whitford, P.C., Noel, J.K., Gosavi, S., Schug, A., Sanbonmatsu, K.Y. and Onuchic, J.N., 2009. An all‐atom structure‐based potential for proteins: bridging minimal models with all‐atom empirical forcefields. Proteins: Structure, Function, and Bioinformatics, 75(2), pp.430-441.</p></li>
</ul>
</div></blockquote>
<dl class="py class">
<dt class="sig sig-object py" id="OpenSMOG.SBM">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">OpenSMOG.</span></span><span class="sig-name descname"><span class="pre">SBM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'SBM'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">time_step</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.002</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">collision_rate</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temperature</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>The <a class="reference internal" href="#OpenSMOG.SBM" title="OpenSMOG.SBM"><code class="xref py py-class docutils literal notranslate"><span class="pre">SBM</span></code></a> class performs Molecular dynamics simulations using structure-based models to investigate a broad range of biomolecular dynamics, including domain rearrangements in proteins, folding and ligand binding in RNA, and large-scale rearrangements in ribonucleoprotein assemblies. In its simplest form, a structure-based model defines a particular structure (usually obtained from X-ray, or NMR, methods) as the energetic global minimum.</p>
<p>The <a class="reference internal" href="#OpenSMOG.SBM" title="OpenSMOG.SBM"><code class="xref py py-class docutils literal notranslate"><span class="pre">SBM</span></code></a> sets the environment to start the molecular dynamics simulations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> (<em>str</em>) – Name used in the output files. (Default value: <em>SBM</em>).</p></li>
<li><p><strong>time_step</strong> (<em>float</em><em>, </em><em>required</em>) – Simulation time step in units of <span class="math notranslate nohighlight">\(\tau\)</span>. (Default value = 0.002).</p></li>
<li><p><strong>collision_rate</strong> (<em>float</em><em>, </em><em>required</em>) – Friction/Damping constant in units of reciprocal time (<span class="math notranslate nohighlight">\(1/\tau\)</span>). (Default value = 1.0).</p></li>
<li><p><strong>r_cutoff</strong> (<em>float</em><em>, </em><em>required</em>) – Cutoff distance to consider non-bonded interactions in units of nanometers. (Default value = 1.5).</p></li>
<li><p><strong>temperature</strong> (<em>float</em><em>, </em><em>required</em>) – Temperature in reduced units. (Default value = 0.5).</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.createReporters">
<span class="sig-name descname"><span class="pre">createReporters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">trajectory</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">energies</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">forces</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interval</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.createReporters"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.createReporters" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates the reporters to provide the output data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>trajectory</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to save the trajectory <em>.dcd</em> file containing the position of the atoms as a function of time. (Default value: <code class="code docutils literal notranslate"><span class="pre">True</span></code>).</p></li>
<li><p><strong>energies</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to save the energies in a <em>.txt</em> file containing five columns comma-delimited. The header of the files shows the information of each collum: #”Step”,”Potential Energy (kJ/mole)”,”Kinetic Energy (kJ/mole)”,”Total Energy (kJ/mole)”,”Temperature (K)”. (Default value: <code class="code docutils literal notranslate"><span class="pre">True</span></code>).</p></li>
<li><p><strong>forces</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to save the potential energy for each applied force in a <em>.txt</em> file containing several columns comma-delimited. The header of the files shows the information of each collum. An example of the header is: #”Step”,”eletrostastic”,”Non-Contacts”,”Bonds”,”Angles”,”Dihedrals”,”contact_1-10-12”. (Default value: <code class="code docutils literal notranslate"><span class="pre">False</span></code>).</p></li>
<li><p><strong>interval</strong> (<em>int</em><em>, </em><em>required</em>) – Frequency to write the data to the output files. (Default value: <code class="code docutils literal notranslate"><span class="pre">10**3</span></code>)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.createSimulation">
<span class="sig-name descname"><span class="pre">createSimulation</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.createSimulation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.createSimulation" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates the simulation context and loads into the OpenMM platform.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.loadGro">
<span class="sig-name descname"><span class="pre">loadGro</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Grofile</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.loadGro"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.loadGro" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the  <em>.gro</em> file format in the OpenMM system platform. The inputs files are generated using SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. Details on how to create the files can be found in the <a class="reference external" href="https://smog-server.org/smog2/">SMOG2 User Manual</a>.
A tutorial on how to generate the inputs files for a C-alpha model simulation can be found <a class="reference external" href="https://opensmog.readthedocs.io">here</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>Grofile</strong> (<em>file</em><em>, </em><em>required</em>) – Initial structure for the MD simulations in <em>.gro</em> file format generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>.  (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.loadSystem">
<span class="sig-name descname"><span class="pre">loadSystem</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Grofile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Topfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Xmlfile</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.loadSystem"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.loadSystem" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the input files in the OpenMM system platform. The input files are generated using SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. Details on how to create the files can be found in the <a class="reference external" href="https://smog-server.org/smog2/">SMOG2 User Manual</a>.
A tutorial on how to generate the inputs files for a C-alpha model simulation can be found <a class="reference external" href="https://opensmog.readthedocs.io">here</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Grofile</strong> (<em>file</em><em>, </em><em>required</em>) – Initial structure for the MD simulations in <em>.gro</em> file format generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>.  (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p></li>
<li><p><strong>Topfile</strong> (<em>file</em><em>, </em><em>required</em>) – Topology <em>.top</em> file format generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. The topology file lists the interactions between the system atoms expect for the “Native Contacts” potential that is provided to OpenSMOG in a <em>.xml</em> file. (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p></li>
<li><p><strong>Xmlfile</strong> (<em>file</em><em>, </em><em>required</em>) – The <em>.xml</em> file that contains the information on the “Contacts” potential. The <em>.xml</em> file is generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code> and accepts Custom Potential forms. (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.loadTop">
<span class="sig-name descname"><span class="pre">loadTop</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Topfile</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.loadTop"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.loadTop" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the  <em>.top</em> file format in the OpenMM system platform. The inputs files are generated using SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. Details on how to create the files can be found in the <a class="reference external" href="https://smog-server.org/smog2/">SMOG2 User Manual</a>.
A tutorial on how to generate the inputs files for a C-alpha model simulation can be found <a class="reference external" href="https://opensmog.readthedocs.io">here</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>Topfile</strong> (<em>file</em><em>, </em><em>required</em>) – Topology <em>.top</em> file format generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. The topology file lists the interactions between the system atoms expect for the “Native Contacts” potential that is provided to OpenSMOG in a <em>.xml</em> file. (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.loadXml">
<span class="sig-name descname"><span class="pre">loadXml</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Xmlfile</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.loadXml"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.loadXml" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the  <em>.xml</em> file format in the OpenMM system platform. The inputs files are generated using SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code>. Details on how to create the files can be found in the <a class="reference external" href="https://smog-server.org/smog2/">SMOG2 User Manual</a>.
A tutorial on how to generate the inputs files for a C-alpha model simulation can be found <a class="reference external" href="https://opensmog.readthedocs.io">here</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>Xmlfile</strong> (<em>file</em><em>, </em><em>required</em>) – The <em>.xml</em> file that contains the information on the “Contacts” potential. The <em>.xml</em> file is generated by SMOG2 software with the flag <code class="code docutils literal notranslate"><span class="pre">-openSMOG</span></code> and accepts Custom Potential forms. (Default value: <code class="code docutils literal notranslate"><span class="pre">None</span></code>).</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.run">
<span class="sig-name descname"><span class="pre">run</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">nsteps</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">report</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interval</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10000</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.run"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.run" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the molecular dynamics simulation.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>nsteps</strong> (<em>int</em><em>, </em><em>required</em>) – Number of steps to perform the simulation. (Default value: <code class="code docutils literal notranslate"><span class="pre">10**7</span></code>)</p></li>
<li><p><strong>report</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to print the simulation progress. (Default value: <code class="code docutils literal notranslate"><span class="pre">True</span></code>).</p></li>
<li><p><strong>interval</strong> (<em>int</em><em>, </em><em>required</em>) – Frequency to print the simulation progress. (Default value: <code class="code docutils literal notranslate"><span class="pre">10**4</span></code>)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.saveFolder">
<span class="sig-name descname"><span class="pre">saveFolder</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">folder</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.saveFolder"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.saveFolder" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the folder path to save data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>folder</strong> (<em>str</em><em>, </em><em>optional</em>) – Folder path to save the simulation data. If the folder path does not exist, the function will create the directory.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="OpenSMOG.SBM.setup_openmm">
<span class="sig-name descname"><span class="pre">setup_openmm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'cuda'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">precision</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'single'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">GPUindex</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'default'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">integrator</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'langevin'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/OpenSMOG.html#SBM.setup_openmm"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#OpenSMOG.SBM.setup_openmm" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets up the parameters of the simulation OpenMM platform.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>platform</strong> (<em>str</em><em>, </em><em>optional</em>) – Platform to use in the simulations. Opitions are <em>CUDA</em>, <em>OpenCL</em>, <em>HIP</em>, <em>CPU</em>, <em>Reference</em>. (Default value: <em>CUDA</em>).</p></li>
<li><p><strong>precision</strong> (<em>str</em><em>, </em><em>optional</em>) – Numerical precision type of the simulation. Opitions are <em>single</em>, <em>mixed</em>, <em>double</em>. (Default value: <em>single</em>). For details check the <a class="reference external" href="http://docs.openmm.org/latest/developerguide/developer.html#numerical-precision">OpenMM Documentation</a>.</p></li>
<li><p><strong>GPUIndex</strong> (<em>str</em><em>, </em><em>optional</em>) – Set of Platform device index IDs. Ex: 0,1,2 for the system to use the devices 0, 1 and 2. (Use only when GPU != default)</p></li>
<li><p><strong>integrator</strong> (<em>str</em>) – Integrator to use in the simulations. Options are <em>langevin</em>,  <em>variableLangevin</em>, <em>verlet</em>, <em>variableVerlet</em> and, <em>brownian</em>. (Default value: <em>langevin</em>).</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="Tutorials/SMOG2_usage.html" class="btn btn-neutral float-right" title="Using SMOG2 - C-alpha and All-Atoms" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="GettingStarted/intro.html" class="btn btn-neutral float-left" title="Introduction" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020-2021 The Center for Theoretical Biological Physics (CTBP) - Rice University &amp; Northeastern University.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>